client_test/liabilityswap_test.go

```diff
@@ -0,0 +1,177 @@
+package client_test
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel/consensus_channel"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+	"github.com/statechannels/go-nitro/client"
+	"github.com/statechannels/go-nitro/client/engine/chainservice"
+	Token "github.com/statechannels/go-nitro/client/engine/chainservice/erc20"
+	"github.com/statechannels/go-nitro/client/engine/messageservice"
+	"github.com/statechannels/go-nitro/client/engine/store"
+	"github.com/statechannels/go-nitro/types"
+	"github.com/stretchr/testify/require"
+)
+
+var liabilitySwapChannelDeposit = big.NewInt(1000000000000000000) // 1 eth in wei
+
+// FIXME:
+// 1. Test fails during the ledger channel funding with this error:
+// 		0x111A00868581f73AB42FEEF67D235Ca09ca1E8db, error in run loop: failed to estimate gas needed: execution reverted: ERC20: transfer amount exceeds balance
+//
+// TODO:
+// 1. Introduce the ability to use a Ledger Channel with custom App (AppDefinition and AppData)
+//   a. Implement direct-transition protocol objective
+//   b. Implement Engine transitionChannel function
+//   c. Update Engine to support pure state channels (decouple from consensus channels, storage, etc.)
+//   d. Check that direct direct-fund/defund protocols support pure state channels
```

Comment:
I disagree with the TODO here, Maybe George could confirm:

### 1. I don't think we will ever perform Trading over direct channels.

The only purpose of direct channels is to fund virtual channels.
If two parties want to trade and have direct-channels, they would create a virtual channel, having 0 intermiediates.

This would discard all tasks related to "pure state channels"

### 2. SerDe of liabilities

At this stage of the protocol, the application is more simple and it's only about sharing collateral (margin)
but I don't see any reason to bother with serialized liabilities as a bi-directional VirtualPaymentApp would do the job.

Reply:
There is nothing about trading here.
1. is about supporting actual generic state channel protocol, so we can use our app and increment turns using go-nitro protocol